contracts/oz/erc-4626/TokenVault.sol

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: Apache-2.0
+
+pragma solidity ^0.8.20;
+
+import "../../erc-20/ERC20Mock.sol";
+
+import "@openzeppelin/contracts/token/ERC20/extensions/ERC4626.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+
+contract TokenVault is ERC4626 {
+
+    // a mapping that checks if a user has deposited the token
+    mapping(address => uint256) public shareHolders;
+
+    constructor(
+        IERC20 _asset,
+        string memory _name,
+        string memory _symbol
+    ) ERC4626(_asset) ERC20(_name, _symbol) {
+       
+    }
+
+    function _deposit(uint256 _assets) public {
+        // checks that the deposited amount is greater than zero.
+        require(_assets > 0, "Deposit is zero");
+        // calling the deposit function from the ERC-4626 library to perform all the necessary functionality
+        deposit(_assets, msg.sender);
```

Comment:
Is this `deposit` method sending assets to msg.sender? If so I think a checks-effects-interactions approach would be better in this case to avoid reentrant attack. Just simply swap the line 30 with this line 28 should fix it I believe.